core/tests/test_views.py

```diff
@@ -177,3 +176,56 @@ def test_add_deeply_nested_comment_valid(client: Client, another_user: User, pos
     assert post.children_count == 10
     assert parent_comment.children_count == 0
     assert parent_comment.parent.children_count == 1
+
+
+@pytest.mark.django_db()
+def test_restricted_community_access(client: Client, restricted_community: Community, user: User) -> None:
+    client.force_login(user)
+    response = client.get(reverse("community-detail", kwargs={"slug": restricted_community.slug}))
+    assert response.status_code == 200
+
+
+def test_create_community_view(client: Client, user: User) -> None:
+    client.force_login(user)
+    url = reverse("community-create")
+    valid_data = {"name": "Test Community", "privacy": "10_PUBLIC", "is_18_plus": True}
+    response = client.post(url, valid_data)
+    assert response.status_code == 302
+    assert Community.objects.filter(name="Test Community").exists()
+
+    invalid_data = {"name": "", "privacy": "INVALID", "is_18_plus": "not_boolean"}
+    response = client.post(url, invalid_data)
+    assert response.status_code == 200
+    assert "form" in response.context
+    assert response.context["form"].errors
```

Comment:
Please add asserts for exact error fields you expect. Here it shows there are errors, but there can be only one error instea of three...
Actually, even assert len(response.context["form"].errors) ==3 would be ok.